futures-util/src/try_stream/mod.rs

```diff
@@ -572,8 +577,44 @@ pub trait TryStreamExt: TryStream {
     /// ```
     #[cfg(feature = "compat")]
     fn compat(self) -> Compat<Self>
-        where Self: Sized + Unpin,
+    where
+        Self: Sized + Unpin,
     {
         Compat::new(self)
     }
+
+    /// Adapter that converts this stream into an [`AsyncRead`].
+    ///
+    /// Note that because `into_async_read` moves the stream, the [`Stream`] type must be
+    /// [`Unpin`]. If you want to use `into_async_read` with a [`!Unpin`](Unpin) stream, you'll
+    /// first have to pin the stream. This can be done by boxing the stream using [`Box::pinned`]
+    /// or pinning it to the stack using the `pin_mut!` macro from the `pin_utils` crate.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(async_await, await_macro)]
+    /// # futures::executor::block_on(async {
+    /// use futures::executor::block_on;
+    /// use futures::future::lazy;
+    /// use futures::stream::{self, StreamExt, TryStreamExt};
+    /// use futures::io::{AsyncRead, AsyncReadExt};
+    /// use std::io::Error;
+    ///
+    /// let stream = stream::iter(vec![Ok(vec![1, 2, 3, 4, 5])]);
+    /// let mut reader = stream.into_async_read();
+    /// let mut buf = Vec::new();
+    ///
+    /// assert!(await!(reader.read_to_end(&mut buf)).is_ok());
+    /// assert_eq!(buf, &[1, 2, 3, 4, 5]);
+    /// # })
+    /// ```
+    #[cfg(feature = "std")]
+    fn into_async_read(self) -> IntoAsyncRead<Self>
+    where
+        Self: Sized + TryStreamExt<Error = std::io::Error> + Unpin,
+        Self::Ok: AsRef<[u8]>,
```

Comment:
ugh it's so annoying that you can have this but `Self::Error = std::io::Error` doesn't work. Someday...